service/src/main/java/bio/terra/externalcreds/services/NihCredentialsSyncService.java

```diff
@@ -0,0 +1,173 @@
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class NihCredentialsSyncService {
+
+  private final ExternalCredsConfig externalCredsConfig;
+  private final FirecloudOrchestrationClient firecloudOrchestrationClient;
+  private final GoogleCloudStorageDAO googleCloudStorageDAO;
+
+  public NihCredentialsSyncService(
+      ExternalCredsConfig externalCredsConfig,
+      FirecloudOrchestrationClient firecloudOrchestrationClient,
+      GoogleCloudStorageDAO googleCloudStorageDAO) {
+    this.externalCredsConfig = externalCredsConfig;
+    this.firecloudOrchestrationClient = firecloudOrchestrationClient;
+    this.googleCloudStorageDAO = googleCloudStorageDAO;
+  }
+
+  /**
+   * Check that all allowlists published by the NIH are still recent enough to be considered valid.
+   * The validity period is set in configs at
+   * `externalcreds.nih-credentials-sync-config.allowlist-validity-duration`
+   *
+   * <p>If `externalcreds.nih-credentials-sync-config.is-fail-closed` is set to true, then invalid
+   * allowlists will be overwritten with empty files, and Firecloud Orch will be told to update the
+   * Sam permissions, which will wipe out all access for the data governed by the allowlist.
+   *
+   * @return if all allowlists are valid or not
+   */
+  public boolean allAllowlistsValid() {
+    var invalidAllowlists = getInvalidAllowlists();
+    if (invalidAllowlists.isEmpty()) {
+      log.info("No NIH allowlists are outside of their validity period");
+      return true;
+    }
+    log.error("NIH allowlists outside of their validity period found!");
+    var now = Instant.now();
+    invalidAllowlists.forEach(
+        b -> {
+          var between = Duration.between(Instant.ofEpochMilli(b.blob().getCreateTime()), now);
+          log.error(
+              "NIH allowlist {} is {} hours and {} minutes old",
+              b.blob().getBlobId().getName(),
+              between.toHoursPart(),
+              between.toMinutesPart());
+        });
+    if (externalCredsConfig.getNihCredentialsSyncConfig().isFailClosed()) {
+      log.warn("Config set to fail-closed. Removing all access for invalid allowlists");
+      var allAllowlistsWiped = failClosed(invalidAllowlists);
+      if (!allAllowlistsWiped) {
+        throw new NihCredentialsSyncException(
+            "SOME PERMISSION-REVOCATION OF ALLOWLISTS FAILED! "
+                + "Check previous logs for lists that need manual intervention.");
+      }
+    } else {
+      log.error(
+          "Config not set to fail-closed. Doing nothing. Someone should definitely check this out");
+    }
+    return false;
+  }
+
+  /**
+   * Take the invalid allowlists, and update permissions so that nobody can access the data.
+   *
+   * @param invalidAllowlists allowlists that are too old to be valid anymore.
+   * @return if all the invalid allowlists could be updated and permissions revoked.
+   */
+  boolean failClosed(List<NihCredentialsBlob> invalidAllowlists) {
```

Comment:
Not sure about current idioms in Java, but I would personally prefer to always specify the scope of a method.  I could see where an engineer changing the scope from `private` to `public` might stop and think.  But simply adding `public` might slip under the radar by accident.  Same applies for all the methods without a specified scope.